### PR TITLE
feat: LEAP-1599: Allow to edit TimelineRegion spans

### PR DESCRIPTION
**Pull request by hlomzik**
Make `TimelineRegion` fields editable, but with new form to display the editor in this case — `start`/`end` are in the `ranges[0]`, not in region itself. New editor is pretty simple and reuses styles from both audio region and image region editors.

Also added an icon for timeline spans.

Screenshot 2024-10-30 at 17 17 25

### PR fulfills these requirements
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

### Describe the reason for change
Timeline spans can only be drawn within visible area and they can't be changed after. So we are adding ability to increase/change them at least via the start/end editable fields.

#### What alternative approaches were there?
Allow to change spans on timeline, but that's much more effort and more complex UX.

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
- [ ] e2e
- [ ] integration
- [ ] unit

### CODE REVIEW

File: web/libs/editor/src/components/SidePanels/DetailsPanel/RegionEditor.tsx [hunk 1]
```diff
@@ -59,18 +52,21 @@ const IconMapping = {
...
+      <Component region={region} />
+    </Block>
+  );
+};
 
-  const changeEndTimeHandler = (value: number) => {
-    region.setProperty("end", value);
-  };
+const RegionProperties = ({ region }: RegionEditorProps) => {
+  const fields: any[] = region.editableFields ?? [];
```
Comment (bmartel) [inline web/libs/editor/src/components/SidePanels/DetailsPanel/RegionEditor.tsx]:
```suggestion
  const fields = region.editableFields ?? [];
```
Does this not already pick up the updated type coming from your newly typed `MSTRegion`?

File: web/libs/editor/src/components/SidePanels/DetailsPanel/RegionEditor.tsx [hunk 2]
```diff
@@ -59,18 +52,21 @@ const IconMapping = {
...
 
-  const renderRegionProperty = () => (
+  return (
     <Elem name="wrapper">
       {region.editorEnabled &&
         fields.map((field: any, i) => {
```
Comment (bmartel) [inline web/libs/editor/src/components/SidePanels/DetailsPanel/RegionEditor.tsx]:
```suggestion
        fields.map((field, i) => {
```

**Review by bmartel — APPROVED** [review]
Looks great, I added a couple suggestions/questions surrounding the type propagation.

**Comment by hlomzik** [comment]
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/11673827398)
> Successfully merged:  delete mode 100644 web/libs/editor/tests/integration/e2e/outliner/filter.cy.ts

**Comment by MihajloHoma** [comment]
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/11699585789)
> Successfully merged:  create mode 100644 label_studio/core/static/templates/video-frame-classification.png

**Review by MihajloHoma — APPROVED** [review]

File: web/libs/editor/src/components/SidePanels/DetailsPanel/RegionEditor.tsx (hunk 1 above)
Comment (hlomzik) [reply web/libs/editor/src/components/SidePanels/DetailsPanel/RegionEditor.tsx]: Thank you! It's actually better without any with type inference!

**Pull request merged**